### DISCUSSION TITLE
Light House or Loco position together with the gas sensor (in I2C communication)

### DISCUSSION THREAD

**Discussion by alinemara86** [bitcraze/discussions#768] [Q&A]
I'm facing difficulties when using the CF2.1 positioning systems (Light House and Loco position) together with the gas sensor (I2C communication). Both positioning systems shows the same error message.

CF2.1* initially operates for a short period of time, but then encounters an initialization issue, resulting in the display of the following error message before terminating the communication:

SYS: The system resumed after a failed assert [WARNING]
SYS: Assert failed at ../src/hal/src/radiolink.c:164
SYS: cfAssertNormalStart [FAIL]

Previously, the same Deck was used in the CF2.0 version and worked without any issues.

*CF2.1 only functions with either the LPS or the gas sensor, not both together.
**CF2.0 supports the entire set, including both the LPS and the gas sensor.

I kindly request guidance and assistance on how to resolve this error message and also achieve the functionality of the LPS with the gas sensor in CF2.1.

**Comment by tobbeanton** [comment]
> SYS: The system resumed after a failed assert [WARNING]
> SYS: Assert failed at ../src/hal/src/radiolink.c:164
> SYS: cfAssertNormalStart [FAIL]

This is normally an indication that the system is overloaded. The communication system does not have enough time and queues build up. Can you try the same thing with communication over USB instead and see if the problem goes away? Solution to this would be to try and load the system less or change task priorities.

Are you running the same base FW on CF2.0 and CF2.1?

**Reply by alinemara86** [reply]
Your observation is quite interesting and has helped me a lot in understanding what is causing the problem. I performed the test using the USB connection and I no longer received that error message. However, there is still a warning stating "Kaman predicative rate low." Is this warning something I should be concerned about?

Regarding CF2.0, it was used with another version of FW.

In the project I'm working on, the gas sensor reading occurs every second. Increasing this reading interval can have a significant impact on research results. Therefore, I would like to request, if possible, some suggestions on how to reduce the system load or change task priorities without compromising the performance and flight stability of the CF. The goal is to allow it to simultaneously perform flight-related tasks and send data from the connected sensor.

**Reply by tobbeanton** [reply]
The ""Kaman predicative rate low." is indication of a temporary high system load but should not be a problem.

The first easy thing to test would be to just comment out the assert in the code on "../src/hal/src/radiolink.c:164". Some radio packets will be lost but might not be a problem.

**Reply by alinemara86** [reply]
Hello Tobias, I sincerely appreciate your attention.
By trying to solve and understand the issue, I performed some tests and obtained the following results:
    1) When I commented out line 164 of the radiolink: the position data appeared blank in cfclient, and the lighthouse was not detected.
    2) I set the sensor read priority to 0, but it had no effect.
    3) I reduced the Lighthouse priority from 3 to 2, but it had no effect too.
    4) I further reduced the Lighthouse priority from 3 to 1, and the positioning system became quite unstable, despite the error messages ceasing.
    During this process, I had some hypotheses. Initially, I thought that implementing the external sensing system using deck drive and app layer might be causing the lack data management.
    Therefore, I implemented the example from the website: https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/development/systemtask/
    When running the latest firmware with the example and the lighthouse simultaneously, the same radio communication overload problem occurred.
I would like to ask if you have any further suggestions on what can be done to manage the system so that data collection and positioning can work simultaneously. Thank you.

**Reply by tobbeanton** [reply]
Perhaps we can replicate your problem then. Hopefully we will have some time this week.

Other thing to try would be to log less data. After connecting to the crazyflie with cfclient go to the "log blocks" tab and untick those you do not need.
